play/src/pusher/models/Space.ts

```diff
@@ -0,0 +1,345 @@
+                    this.isFilterTargeted(spaceFilter, subMessage, user)
+                );
+                if (filtersTargeted.length > 0) {
+                    filtersTargeted.forEach((spaceFilter) => {
+                        if (subMessage.message?.$case === "addSpaceUserMessage") {
+                            subMessage.message.addSpaceUserMessage.filterName = spaceFilter.filterName;
+                        } else if (subMessage.message?.$case === "updateSpaceUserMessage") {
+                            subMessage.message.updateSpaceUserMessage.filterName = spaceFilter.filterName;
+                        } else if (subMessage.message?.$case === "removeSpaceUserMessage") {
+                            subMessage.message.removeSpaceUserMessage.filterName = spaceFilter.filterName;
+                        }
+                        watcher.emitInBatch(subMessage);
+                    });
+                } else {
+                    watcher.emitInBatch(subMessage);
+                }
+            });
+    }
+
+    public notifyMe(watcher: ExSocketInterface, subMessage: SubMessage, user: SpaceUser | undefined = undefined) {
+        watcher.emitInBatch(subMessage);
+    }
+
+    private isWatcherTargeted(
+        watcher: ExSocketInterface,
+        subMessage: SubMessage,
+        user: SpaceUser | undefined = undefined
+    ) {
+        const filtersOfThisSpace = watcher.spacesFilters.filter((spaceFilters) => spaceFilters.spaceName === this.name);
+        return (
+            filtersOfThisSpace.length === 0 ||
+            filtersOfThisSpace.filter((spaceFilter) => this.isFilterTargeted(spaceFilter, subMessage, user)).length > 0
+        );
+    }
+
+    private isFilterTargeted(
+        spaceFilter: SpaceFilterMessage,
+        subMessage: SubMessage,
+        user: SpaceUser | undefined = undefined
+    ) {
+        if (spaceFilter.spaceName === this.name) {
+            let user_ = user;
+            if (subMessage.message?.$case === "addSpaceUserMessage") {
+                user_ = subMessage.message.addSpaceUserMessage.user;
+            } else if (subMessage.message?.$case === "updateSpaceUserMessage") {
+                // I'm sure user uuid is existing
+                user_ = this.users.get(subMessage.message.updateSpaceUserMessage.user?.uuid as string);
+            }
+            if (user_) {
+                return this.filterOneUser(spaceFilter, user_);
+            }
+        }
+        return false;
+    }
+
+    public filter(spaceFilter: SpaceFilterMessage) {
+        return [...this.users.values()].filter((user) => this.filterOneUser(spaceFilter, user));
+    }
+
+    private filterOneUser(spaceFilters: SpaceFilterMessage, user: SpaceUser) {
+        if (spaceFilters.filter?.$case === "spaceFilterContainName") {
+            const spaceFilterContainName = spaceFilters.filter.spaceFilterContainName;
+            if (user.name.includes(spaceFilterContainName.value)) {
```

Comment:
Maybe switch to lowercase to do a case insensitive comparison.

Pro-tip: lowercasing can be long so you might want to store the "user.lowerCaseName"